src/Action.hs

```diff
@@ -129,6 +130,8 @@ class (Monad r, Monad m, MonadError ActionExcept m) => ActionPersist r m | m ->
     -- | Run "Persistent" computation in the 'Action' monad.
     persistent :: r a -> m a  -- TODO: rename to atomic
 
+class (ActionPersist AQuery m, ActionPersist AUpdate m) => ActionPersistM m
```

Comment:
Using `type ActionPersistM m = (ActionPersist AQuery m, ActionPersist AUpdate m)` is better IMO, in particular this avoids the need for empty instances.